app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -16,19 +17,41 @@ const METRICS = [
   { text: "Depth", key: "depth" },
 ];
 
+const DEFAULT_METADATA_TYPE = ["collection_location"];
+
 export default class AMRHeatmapVis extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       nodeHoverInfo: null,
       tooltipLocation: null,
+      columnMetadataLegend: null,
+      addMetadataTrigger: null,
+      selectedMetadata: new Set(DEFAULT_METADATA_TYPE),
     };
 
     this.heatmap = null;
   }
 
   componentDidMount() {
+    this.processSampleData();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { selectedMetadata } = this.state;
+    if (this.props !== prevProps || this.heatmap === null) {
```

Comment:
I didn't catch this before, but comparing all props is bad practice. You should just compare the props that actually affect the rendering of the heatmap.

Reply:
It's bad practice because you might run extra code unnecessarily.